src/app/api/index.ts

```diff
@@ -319,44 +322,82 @@ class Client implements API {
+                return WORK_SECTIONS[assignment.workSectionId];
             }
         }
+        else {
+            return "Assignment Template";
+        }
 
-        //add default recurrence value if nothing was selected or partial value was selected
-        if(!newTemplate.recurrenceInfo){
-            newTemplate.recurrenceInfo = [DEFAULT_RECURRENCE];
+    }
+
+    private setAssignmentTemplateDefaultRecurrenceInfo(template: Partial<SheriffAssignmentTemplate>): RecurrenceInfo[] {
+        if (!template.recurrenceInfo || template.recurrenceInfo.length === 0) {
+            return [DEFAULT_RECURRENCE];
         }
-        else{
-            newTemplate.recurrenceInfo.forEach(element => {
-                if(!element.startTime){
+        else {
+            template.recurrenceInfo.forEach(element => {
+                if (!element.startTime) {
                     element.startTime = DEFAULT_RECURRENCE.startTime;
                 }
-                if(!element.endTime){
+                if (!element.endTime) {
                     element.endTime = DEFAULT_RECURRENCE.endTime;
                 }
-                if(!element.days){
+                if (!element.days) {
                     element.days = DEFAULT_RECURRENCE.days;
                 }
             });
+            return template.recurrenceInfo;
+        }
+    }
+
+    async createAssignmentTemplate(newTemplate: Partial<SheriffAssignmentTemplate>): Promise<SheriffAssignmentTemplate> {
+        await randomDelay();
+
+        if (!newTemplate || !newTemplate.assignment) {
+            throw new Error("Incomplete new assignment template.")
         }
 
-        defaultAssignmentTemplates.push(newTemplate);
-        return newTemplate;
+        let assignment = newTemplate.assignment;
+
+        //This is a hack to create a unique id for a new assignment template
+        newTemplate.id = ASSIGNMENT_TEMPLATE_ID;
+        assignment.id = ASSIGNMENT_TEMPLATE_ID;
+        ASSIGNMENT_TEMPLATE_ID++;
+
+        assignment.title = this.setAssignmentTemplateTitle(assignment);
+
+        //add default recurrence value if nothing was selected or partial value was selected
+        newTemplate.recurrenceInfo = this.setAssignmentTemplateDefaultRecurrenceInfo(newTemplate);
+
+        defaultAssignmentTemplates.push(newTemplate as SheriffAssignmentTemplate);
+        return newTemplate as SheriffAssignmentTemplate;
+    }
+
+    async editAssignmentTemplate(updatedAssignmentTemplate: SheriffAssignmentTemplate): Promise<SheriffAssignmentTemplate> {
+        await randomDelay();
+        let assignment = updatedAssignmentTemplate.assignment;
+        
+        assignment.title =  this.setAssignmentTemplateTitle(assignment);
+        
+        //add default recurrence value if nothing was selected or partial value was selected
+        updatedAssignmentTemplate.recurrenceInfo = this.setAssignmentTemplateDefaultRecurrenceInfo(updatedAssignmentTemplate);
```

Comment:
So here you're mixing the get/set pattern.  i.e. your passing in and mutating `updatedAssignmentTemplate` but then returning the mutated `recurrenceInfo`.  See above ⬆️